skimage/filters/_gabor.py

```diff
@@ -72,35 +254,63 @@ def gabor_kernel(frequency, theta=0, bandwidth=1, sigma_x=None, sigma_y=None,
     >>> io.imshow(gk.real)  # doctest: +SKIP
     >>> io.show()           # doctest: +SKIP
     """
-    if sigma_x is None:
-        sigma_x = _sigma_prefactor(bandwidth) / frequency
-    if sigma_y is None:
-        sigma_y = _sigma_prefactor(bandwidth) / frequency
+    # handle deprecation
+    message = ('Using deprecated, 2D-only interface to gabor_kernel. '
+               'This interface will be removed in scikit-image 0.16. Use '
+               'gabor_kernel(frequency, sigma=(sigma_x, sigma_y)).')
+
+    if sigma_y is not None:
+        warn(message)
+        if 'sigma_x' in kwargs:
+            sigma = (sigma_y, kwargs['sigma_x'])
+        else:
+            sigma = (sigma_y, sigma)
+
+    # handle translation
+    if not isinstance(theta, coll.Iterable):
+        theta = (theta,) * (ndim - 1)
 
-    x0 = np.ceil(max(np.abs(n_stds * sigma_x * np.cos(theta)),
-                     np.abs(n_stds * sigma_y * np.sin(theta)), 1))
-    y0 = np.ceil(max(np.abs(n_stds * sigma_y * np.cos(theta)),
-                     np.abs(n_stds * sigma_x * np.sin(theta)), 1))
-    y, x = np.mgrid[-y0:y0 + 1, -x0:x0 + 1]
+    if not isinstance(sigma, coll.Iterable):
+        sigma = np.array([sigma] * ndim)
+    else:
+        sigma = np.append(sigma, [None] * (ndim - len(sigma)))
+    default_sigma = _sigma_prefactor(bandwidth) / frequency
+    sigma[(sigma == None).nonzero()] = default_sigma  # noqa
```

Comment:
No need for the `nonzero` call. But see above: `default_sigma` should only be used when *no* sigma has been given.

Reply:
@jni I chose this so that users could manually specify when they wanted `sigma` to be automatically calculated for them, ie if I wanted my third axis only to be automatically calculated for me, I could specify `sigma=(5, 10, None)` which would be equivalent to `sigma=(5, 10, default_sigma)`.